internal/controller/pkg/revision/runtime_defaults.go

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright 2023 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package revision
+
+import (
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var (
+	replicas                 = int32(1)
+	runAsUser                = int64(2000)
+	runAsGroup               = int64(2000)
+	allowPrivilegeEscalation = false
+	privileged               = false
+	runAsNonRoot             = true
+)
+
+func defaultServiceAccount(name string) *corev1.ServiceAccount {
+	return &corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			// It is possible to override the name of the service account in the
+			// RuntimeConfig. So, we define it as a default here.
+			Name: name,
+		},
+	}
+}
+
+func defaultDeployment(name string) *appsv1.Deployment {
```

Comment:
wouldn't it make sense to have this return something like a `DeploymentBuilder` which could be the receiver for all the functions in internal/controller/pkg/revision/runtime_override_options.go, such as `DeploymentWithName`, doing the same for Service and ServiceAccount?

Reply:
Yes, it could be doable, but I wanted to be consistent with the pattern we're following in the project: functional options instead of builders.

Also, please note, that this function no longer exists in the latest version in https://github.com/crossplane/crossplane/pull/4744